CI: `staging` fixture not working

[Client support for Rekor V2: sigstore-python](https://github.com/sigstore/rekor-tiles/issues/289)


**Description**

<!--Tell us about the bug. What went wrong? How can we reproduce it? What should the correct behavior be?
-->

The `staging` fixture needs updating, after merging #1363. It's strange that the tests didn't fail within the PR's checks, only after merging. Tests also fail locally.

https://github.com/sigstore/sigstore-python/actions/runs/15190933113/job/42723267534#step:6:217

```log
==================================== ERRORS ====================================
_________ ERROR at setup of test_sign_rekor_entry_consistent[staging] __________

pytestconfig = <_pytest.config.Config object at 0x7f69195a64d0>, env = 'staging'

    @pytest.fixture
    def sign_ctx_and_ident_for_env(
        pytestconfig,
        env: str,
    ) -> tuple[type[SigningContext], type[IdentityToken]]:
        if env == "staging":
>           ctx_cls = SigningContext.staging
E           AttributeError: type object 'SigningContext' has no attribute 'staging'

env        = 'staging'
pytestconfig = <_pytest.config.Config object at 0x7f69195a64d0>

test/unit/conftest.py:192: AttributeError
```

**Version**

<!--What version are you using? e.g. `cosign version`...
-->

@main 916c2c5ac2e307637379a27eef71adf6d2b0272d

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -36,6 +36,10 @@ All versions prior to 0.9.0 are untracked.
   [sigstore/timestamp-authority](https://github.com/sigstore/timestamp-authority)
   [#1377](https://github.com/sigstore/sigstore-python/pull/1377)
 
+* Tests: Update the `staging` and `sign_ctx_and_ident_for_env` fixtures to use the new methods
+  for generating a `SigningContext`.
+  [#1409](https://github.com/sigstore/sigstore-python/pull/1409)
+
 ### Changed
 
 * API:

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -36,7 +36,7 @@ All versions prior to 0.9.0 are untracked.
   [sigstore/timestamp-authority](https://github.com/sigstore/timestamp-authority)
   [#1377](https://github.com/sigstore/sigstore-python/pull/1377)
 
-* Tests: Update the `staging` and `sign_ctx_and_ident_for_env` fixtures to use the new methods
+* Tests: Updated the `staging` and `sign_ctx_and_ident_for_env` fixtures to use the new methods
   for generating a `SigningContext`.
   [#1409](https://github.com/sigstore/sigstore-python/pull/1409)
 

--- a/test/unit/conftest.py
+++ b/test/unit/conftest.py
@@ -37,6 +37,7 @@ from tuf.ngclient import FetcherInterface, updater
 from sigstore._internal import tuf
 from sigstore._internal.rekor import _hashedrekord_from_parts
 from sigstore._internal.rekor.client import RekorClient
+from sigstore._internal.trust import ClientTrustConfig
 from sigstore._utils import sha256_digest
 from sigstore.models import Bundle
 from sigstore.oidc import _DEFAULT_AUDIENCE, IdentityToken
@@ -188,10 +189,20 @@ def sign_ctx_and_ident_for_env(
     pytestconfig,
     env: str,
 ) -> tuple[type[SigningContext], type[IdentityToken]]:
+    """
+    Returns a SigningContext and IdentityToken for the given environment.
+    The SigningContext is behind a callable so that it may be lazily evaluated.
+    """
     if env == "staging":
-        ctx_cls = SigningContext.staging
+
+        def ctx_cls():
+            return SigningContext.from_trust_config(ClientTrustConfig.staging())
+
     elif env == "production":
-        ctx_cls = SigningContext.production
+
+        def ctx_cls():
+            return SigningContext.from_trust_config(ClientTrustConfig.production())
+
     else:
         raise ValueError(f"Unknown env {env}")
 
@@ -205,7 +216,14 @@ def sign_ctx_and_ident_for_env(
 
 @pytest.fixture
 def staging() -> tuple[type[SigningContext], type[Verifier], IdentityToken]:
-    signer = SigningContext.staging
+    """
+    Returns a SigningContext, Verifier, and IdentityToken for the staging environment.
+    The SigningContext and Verifier are both behind callables so that they may be lazily evaluated.
+    """
+
+    def signer():
+        return SigningContext.from_trust_config(ClientTrustConfig.staging())
+
     verifier = Verifier.staging
 
     # Detect env variable for local interactive tests.

